sunpy/map/sources/suvi.py

```diff
@@ -0,0 +1,77 @@
+"""SUVI Map subclass definitions"""
+from __future__ import absolute_import, print_function, division
+#pylint: disable=W0221,W0222,E1101,E1121
+
+__author__ = "Jack Ireland"
+__email__ = "jack.ireland@nasa.gov"
+
+import matplotlib.pyplot as plt
+
+from astropy.visualization.mpl_normalize import ImageNormalize
+from astropy.visualization import AsinhStretch
+
+from sunpy.map import GenericMap
+from sunpy.map.sources.source_type import source_stretch
+
+__all__ = ['SUVIMap']
+
+
+class SUVIMap(GenericMap):
+    """SUVI Image Map.
+
+    The Solar Ultraviolet Imager (SUVI) is a normal-incidence Cassegrain telescope
```

Comment:
So you got me so excited by this that I wrote up a set of requirements for what I consider a proper doc string. Here they are

Must provide the following information if available

* the name of the mission and name of the instrument including the name of the institution that built it
* a short description of the instrument (e.g. Cassegrain reflector, Wolter-1 grazing incidence x-ray, coronagraph) including the type of detector
* a description of the platform (e.g. satellite in 28 deg inclined orbit, a telescope on the summit of Mauna Kea, Hawaii)
* a description of the purpose or science goals of the instrument.
* a list of all wavelength(s) or passbands in appropriate units
* a description of the emission processes which dominate in those passbands
* the field of view and resolution (e.g. angular resolution)
* the cadence of the measurements
* a description of the operational concept (e.g. operates 24/7, observes from 7 am to 5 pm UT)
* the start and end of the data set

A list of references must be provided which link to the following resources, if available,

* the official mission page 
* the official instrument page
* relevant wikipedia pages
* relevant user guide(s)
* reference to the mission paper
* reference to the instrument paper
* information to interpret metadata keywords such as FITS header reference
* link to the data archive

I then went ahead and rewrote your doc string as an exercise. Here it is

```
The Solar Ultraviolet Imager (SUVI) is a normal-incidence Cassegrain EUV telescope
onboard the latest of the Geostationary Operational Environmental Satellite (GOES) missions (GOES-16).
It is similar to Atmospheric Imaging Assembly (AIA).
It operates in geostationary orbit above the Americas at 75.2 degree W.
It's primary purpose is to support NOAA's goal to characterize solar features and detect
events that lead to space weather.
It uses a filter wheel to image the Sun in six EUV wavelength corresponding to known coronal emission lines: 
* 9.4 nm (FeXVIII)
* 13.1 nm (FeXXI)
* 17.1 nm (FeIX/X)
* 19.5 nm (FeXII)
* 28.4 nm (FeXV) 
* 30.4 nm (HeII)
The focal plane consists of a CCD detector with 1280 x 1280 pixels.
The plate scale is 2.5 arcsec per pixel.
The field of view is therefore almost twice the size of the Sun (53 arcmin) 
and extends out to 1.6 solar radii in the horizontal direction and 2.3 solar radii in the diagonal.
It provides observations in each wavelength at multiple exposure times every 4 minutes.

It began operating on ???.
```

Reply:
If this looks good to people then I'll submit a PR to the docs with these requirements and people can also suggest things that may be missing.

Reply:
@ayshih can you ask Dan when is their official start of operations date. Also I can't seem to find their data. Is it officially released?

Reply:
There may not be a simple answer since they have occasional observations well before becoming fully operational, but I can see what Dan says.

I just posted data links in a top-level comment.

Reply:
@ehsteve make the PR.

Reply:
Also, your requirements should be documented somewhere (developer's guide?) and generalized to non-map sources.

Reply:
done